src/audio_excl.rs

```diff
@@ -121,24 +109,24 @@ pub fn init_map(
     subdirs: Vec<String>,
     timeout: Duration,
 ) -> io::Result<HashMap<String, FileLister>> {
-    let mut ret: HashMap<String, FileLister> = HashMap::new();
-    for subdir in subdirs {
-        let lister = FileLister::new(base_dir.join(&subdir), &timeout)?;
-        ret.insert(subdir, lister);
-    }
-    Ok(ret)
+    subdirs
+        .into_iter()
+        .map(|subdir| (base_dir.join(&subdir), subdir))
+        .map(|(resolved_subdir, subdir)| Ok((subdir, FileLister::new(resolved_subdir, timeout)?)))
+        .collect()
```

Comment:
Because we keep the length estimate of the `Vec` iterator, the `HashMap` can initialize itself with the correct capacity to avoid reallocating while constructing the `HashMap`.